includes/class-sensei-usage-tracking.php

```diff
@@ -0,0 +1,321 @@
+			'teacher_count' => self::get_teacher_count(),
+		) );
+
+		$resp = self::send_event( 'stats_log', $usage_data );
+
+		// Send a dummy event at the same time to enable using Tracks funnels,
+		// which requires a minimum of two event steps.
+		$resp = self::send_event( 'sensei_dummy_stats_log' );
+
+		return $resp;
+	}
+
+	/**
+	 * Get the number of teachers.
+	 *
+	 * @return int Number of teachers.
+	 **/
+	public static function get_teacher_count() {
+		$teacher_query = new WP_User_Query( array( 'role' => 'teacher' ) );
+
+		return $teacher_query->total_users;
+	}
+
+	/**
+	 * Get the total number of learners enrolled in at least one course.
+	 *
+	 * @return int Number of learners.
+	 **/
+	public static function get_learner_count() {
+		$learner_count = 0;
+		$args['fields'] = array( 'ID' );
+		$user_query = new WP_User_Query( $args );
+		$learners = $user_query->get_results();
+
+		foreach( $learners as $learner ) {
+			$course_args = array(
+				'user_id' => $learner->ID,
+				'type' => 'sensei_course_status',
+				'status' => 'any',
+			);
+
+			$course_count = Sensei_Utils::sensei_check_for_activity( $course_args );
+
+			if ( $course_count > 0 ) {
+				$learner_count++;
+			}
+		}
+
+		return $learner_count;
+	}
+
+	/**
+	 *
+	 * Register usage submenu page.
+	 **/
+	function register_usage_submenu_page() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		add_submenu_page(
+			'sensei',
+			__( 'Usage Tracking', 'woothemes-sensei' ),
+			__( 'Usage Tracking', 'woothemes-sensei' ),
+			'manage_options',
+			'sensei_usage_tracking',
+			array( $this, 'render_usage_tracking_page' )
+		);
+	}
+
+	function render_usage_tracking_page() {
+		?>
+		<div class="wrap">
+
+			<div id="icon-woothemes-sensei" class="icon32"><br></div>
+			<h1><?php _e('Sensei Usage Tracking', 'woothemes-sensei'); ?></h1>
+			<form method="post" action="" name="update-sensei" class="upgrade">
+				<p>
+					<input id="update-sensei" type="hidden" value="send_data" name="usage_tracking_action">
+					<input id="send-data" class="button button-primary" type="submit" value="Send Data Now">
+					<?php wp_nonce_field( 'send_data_nonce' ); ?>
+				</p>
+			</form>
+		</div>
+		<?php
+	}
+
+	function add_setting_field( $fields ) {
+		// default-settings
+		$fields[ self::$usage_tracking_setting_name ] = array(
+			'name' => __( 'Enable usage tracking', 'woothemes-sensei' ),
+			'description' => __( 'Allow Sensei to anonymously track plugin usage**. No sensitive data is tracked.', 'woothemes-sensei' ),
+			'type' => 'checkbox',
+			'default' => false,
+			'section' => 'default-settings'
+		);
+
+		return $fields;
+	}
+
+	function maybe_display_tracking_opt_in() {
+		$opt_in_displayed = (bool) get_option( 'sensei_usage_tracking_opt_in_display' );
+		$user_tracking_enabled = (bool) get_option( self::$usage_tracking_setting_name );
+
+		if ( false && ! $user_tracking_enabled && ! $opt_in_displayed ) { ?>
```

Comment:
Was this `false` meant to be included in the PR?

Reply:
This bit doesn't actually work, even if I take `false` out, because Sensei shows a welcome screen that clobbers the notice. I'm currently trying to figure out what the UI for this should look like on activation.